src/campaign_server/addon_utils.cpp

```diff
@@ -92,23 +92,43 @@ std::string format_addon_feedback_url(const std::string& format, const config& p
 	return std::string();
 }
 
+void support_translation(config& addon, std::string locale_id)
+{
+	if(!locale_id.empty()) {
+		config& locale = addon.find_child("translation", "language", locale_id);
+		if(locale) {
+			locale["supported"] = true;
+		}
+		else {
+			addon.add_child("translation")["language"] = locale_id; //Doing this to circumvent a weird validation bug
+			addon.find_child("translation", "language", locale_id)["supported"] = true;
+		}
+	}
+}
+
 void find_translations(const config& base_dir, config& addon)
 {
 	for(const config& file : base_dir.child_range("file")) {
 		const std::string& fn = file["name"].str();
 		if(filesystem::ends_with(fn, ".po")) {
-			addon.add_child("translation")["language"] = filesystem::base_name(fn, true);
+			support_translation(addon, filesystem::base_name(fn, true));
 		}
 	}
 
 	for(const config &dir : base_dir.child_range("dir"))
 	{
 		if(dir["name"] == "LC_MESSAGES") {
-			addon.add_child("translation")["language"] = base_dir["name"];
+			support_translation(addon, base_dir["name"]);
 		} else {
 			find_translations(dir, addon);
 		}
 	}
+
+	for(config& locale : addon.child_range("translation")) {
+		if(!locale["supported"].to_bool(false)) {
+			locale["supported"] = false;
+		}
+	}
```

Comment:
I assume this is intended to prevent people from inputting crap into `translation.supported` values, but it just makes `false` values become... `false`. It would make more sense to go over all `translation` nodes in the caller (to make the semantics of `find_translation` less oblique) and **delete** the `supported` attributes as well **all unrecognized data** (including other WML children nodes) in a first pass at the start of the function, before looking at the archive's contents.

The reason for deleting all unrecognized data is that otherwise it could well become possible for add-on authors to break clients in the future. As well as the fact that they could just abuse this to deliver extraneous data that the client won't process and increase the size of the `[request_campaign_list]` response for no good reason.

Reply:
Accidentally posted this with obsolete pointers the first time around. I want to re-emphasize that `find_translation` shouldn't be in charge of sanitizing the input, that should be the caller's responsibility.

Reply:
Good idea, thanks! I'll make it.